### PR TITLE
feat(konnect): add KongCredentialAPIKey reconciler

### LINKED ISSUE #619: Konnect entities: Credentials - Key Auth
### Problem statement

Introduce `KongCredentialAPIKey`.

API details: https://kong-platform-api.netlify.app/konnect/control-planes-config/v2/openapi.yaml/#tag/API-keys

### Proposed solution

- Introduce `KongCredentialAPIKey` CRD in https://github.com/Kong/kubernetes-configuration
  - All Kong configuration CRDs will live in that repository to prevent import cycles between KGO and KIC
  - This CRD will contain a `KongConsumer` reference. 
  - ControlPlane reference will be inferred from `KongConsumer` reference
  - Only references that refer to Konnect will result in reconciliation against Konnect.

- Implement a reconciler which will use that CRD.

- This reconciler should be able to be disabled via CLI flag / env variable.
  - There should exist only 1 CLI flag / env variable for all Kong configuration entities like Consumers, Services, Routes etc.

### Acceptance criteria

- [x] KGO can create, delete and update DataPlane's client certificates in Konnect based on the CRD spec.
- [x] Status fields of this CRD are filled with information such as
  - [x] Indication whether the entity in Konnect has been created via a `Programmed` status condition

### PR DESCRIPTION
**Pull request by pmalek**
**What this PR does / why we need it**:

~Requires: https://github.com/Kong/kubernetes-configuration/pull/94~

**Which issue this PR fixes**

Fixes #619

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect significant changes

### CODE REVIEW

File: controller/konnect/index_credentials_apikey.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+		{
+			IndexObject:  &configurationv1alpha1.KongCredentialAPIKey{},
+			IndexField:   IndexFieldKongCredentialAPIKeyReferencesKongConsumer,
+			ExtractValue: kongKongCredentialAPIKeyReferencesConsumer,
+		},
+	}
+}
+
+// kongKongCredentialAPIKeyReferencesConsumer returns the name of referenced Consumer.
+func kongKongCredentialAPIKeyReferencesConsumer(obj client.Object) []string {
```
Comment (randmonkey) [inline controller/konnect/index_credentials_apikey.go]: Would the name `kongKongCredentialAPIKeyReferencesConsumer` be redundant?

**Review by randmonkey — COMMENTED** [review]
Generally looks good.

File: controller/konnect/index_credentials_apikey.go (hunk 1 above)
Comment (pmalek) [reply controller/konnect/index_credentials_apikey.go]: What do you mean by redundant here? 🤔 Can you elaborate?

Comment (randmonkey) [reply controller/konnect/index_credentials_apikey.go]: I mean does the name `KongCredentialAPIKeyReferencesConsumer` look better? two consecutive `kong`s looks redundant.

Comment (pmalek) [reply controller/konnect/index_credentials_apikey.go]: Ah this 😅 I haven't noticed this for some reason. 5285f498d6439257e71b6833b0f6de04569cc3b7 fixes that.

**Review by randmonkey — APPROVED** [review]

**Review by czeslavo — APPROVED** [review]

**Pull request merged**